lib/ansible/modules/system/aix_lvol.py

```diff
@@ -0,0 +1,362 @@
+        if match is not None:
+            size = int(match.group(1))
+            continue
+
+        match = re.search("PP SIZE:\s+(\d+)", line)
+        if match is not None:
+            pp_size = int(match.group(1))
+            continue
+
+        match = re.search("FREE PP.*\((\d+)", line)
+        if match is not None:
+            free = int(match.group(1))
+            continue
+
+    return {'name': name, 'size': size, 'free': free, 'pp_size': pp_size}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            vg=dict(required=True),
+            lv=dict(required=True),
+            lv_type=dict(required=False),
+            size=dict(type='str'),
+            opts=dict(type='str'),
+            copies=dict(type='str'),
+            state=dict(choices=["absent", "present"], default='present'),
+            policy=dict(choices=["maximum", "minimum"], default='maximum'),
+            pvs=dict(type='str')
+        ),
+        supports_check_mode=True,
+    )
+
+    vg = module.params['vg']
+    lv = module.params['lv']
+    lv_type = module.params['lv_type']
+    size = module.params['size']
+    opts = module.params['opts']
+    copies = module.params['copies']
+    policy = module.params['policy']
+    state = module.params['state']
+    pvs = module.params['pvs']
+
+    if pvs is None:
+        pvs = ""
+    else:
+        pvs = pvs.replace(",", " ")
+
+    if opts is None:
+        opts = ""
+
+    if copies is None:
+        copies = "1"
+
+    if lv_type is None:
+        lv_type = "jfs2"
+
+    if policy == "maximum":
+        lv_policy="x"
+    else:
+        lv_policy="m"
+
+    # Add --test option when running in check-mode
+    if module.check_mode:
+        test_opt = 'echo '
+    else:
+        test_opt = ''
+
+
+    # Get information on volume group requested
+    lsvg_cmd = module.get_bin_path("lsvg", required=True)
+    rc, vg_info, err = module.run_command(
+        "%s %s" % (lsvg_cmd, vg))
+
+    if rc != 0:
+        if state == 'absent':
+            module.exit_json(changed=False, msg="Volume group %s does not exist." % vg)
+        else:
+            module.fail_json(msg="Volume group %s does not exist." % vg, rc=rc, err=err)
+
+    this_vg = parse_vg(vg_info)
+
+
+    if size is not None:
+        # Calculate pp size and round it up based on pp size.
+        lv_size = round_ppsize(convert_size(size), base=this_vg['pp_size'])
+
+    # Get information on logical volume requested
+    lslv_cmd = module.get_bin_path("lslv", required=True)
+    rc, lv_info, err = module.run_command(
+        "%s %s" % (lslv_cmd, lv))
+
+    if rc != 0:
+        if state == 'absent':
+            module.exit_json(changed=False, msg="Logical Volume %s does not exist." % lv)
+
+    changed = False
+
+    this_lv = parse_lv(lv_info)
+
+    if state == 'present' and not size:
+        if this_lv is None:
+            module.fail_json(msg="No size given.")
+
+    msg=''
```

Comment:
Instead of setting this out here and doing the `exit_json` out of scope, can you please use `exit_json` right in the various `rc==0` blocks?